libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesCacheManager.kt

```diff
@@ -21,14 +22,24 @@ internal class RouteAlternativesCacheManager(
         }
     }
 
+    @VisibleForTesting
+    internal companion object {
+        internal const val CACHE_SIZE = 3
+    }
+
     init {
         navigationSession.registerNavigationSessionStateObserver(sessionStateObserver)
     }
 
     fun push(alternatives: List<DirectionsRoute>) {
-        cachedAlternatives.addAll(alternatives)
+        if (cachedAlternatives.size == CACHE_SIZE) {
+            cachedAlternatives.remove()
+        }
+        cachedAlternatives.add(alternatives)
     }
 
     fun areAlternatives(routes: List<DirectionsRoute>): Boolean =
-        routes.map { it in cachedAlternatives }.any { it }
+        cachedAlternatives.flatten().let { cacheList ->
+            return@let routes.map { it in cacheList }.any { it }
+        }
```

Comment:
what does the method do? does it return true if any of `DirectionsRoute` from the `routes` is present in any of cached `alternatives` ?

If so, maybe this one is easier to read?
```suggestion
        cachedAlternatives.any { cachedAlternative ->
            routes.any { cachedAlternative.contains(it)  }
        }
```

Reply:
yeah, looks good 👍